File: Volume0_0090/main.c
//
//  main.c
//  Volume0_0090
//
//  Created by slangsoft on 2015/01/01.
//  Copyright (c) 2015 Slang Software. All rights reserved.
//

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

/*----------------------------------*/
/* constant declaration             */
/*----------------------------------*/
#define NUM_EXIT    0
#define NUM_MAX_N   100
#define NUM_MIN_XY  0
#define NUM_MAX_XY  10
#define NUM_DIST    2
#define MSG_FST     "\nType the number of datasets (or '0' when done) >>>\n"
#define MSG_SCD     "\nType the dataset >>>\n"
#define MSG_ERR_N   "\nError!\nThe value of N must be 100 or less.\n"
#define MSG_ERR_XY  "\nError!\nThe value of X and Y must be 0 or more and 10 or less.\nPlease retype.\n"

/*----------------------------------*/
/* macro declaration                */
/*----------------------------------*/

/*----------------------------------*/
/* global variable declaration      */
/*----------------------------------*/

/*----------------------------------*/
/* functions prototype declaration  */
/*----------------------------------*/

/*----------------------------------*/
/* structural definition            */
/*----------------------------------*/
struct StdDataset
{
    double  x;
    double  y;
};

/*--------------------------------------*/
/* Function Name : mystrtod             */
/* Argument      : char   *str          */
/*                 double *x            */
/*                 double *y            */
/* Outputs       : -                    */
/* Author        : slangsoft            */
/* Creation Date : 2015/01/01           */
/*--------------------------------------*/
void mystrtod( char *str, double *x, double *y ) {
    char *ends;     /* 2nd arg of strtod() */
    char *str_x;
    char *str_y;
    
    str_x = strtok( str , "," );
    str_y = strtok( NULL, "," );
    
    *x = strtod( str_x, &ends );
    *y = strtod( str_y, &ends );
}

/*--------------------------------------*/
/* Function Name : dist                 */
/* Argument      : double a1            */
/*                 double a2            */
/*                 double b1            */
/*                 double b2            */
/* Output        : double dist          */
/* Author        : slangsoft            */
/* Creation Date : 2015/01/01           */
/*--------------------------------------*/
double dist( double a1, double a2, double b1, double b2 ) {
    double a    = 0;
    double b    = 0;
    double dist = 0;
    
    if( a1 > a2 ) {
        a = a1 - a2;
    }
    else {
        a = a2 - a1;
    }
    
    if( b1 > b2 ) {
        b = b1 - b2;
    }
    else {
        b = b2 - b1;
    }
    
    dist = sqrt( pow( a, 2 ) + pow( b, 2 ) );
    return dist;
}

/*--------------------------------------*/
/* Function Name : overlaps             */
/* Argument      : int n                */
/* Output        : -                    */
/* Author        : slangsoft            */
/* Creation Date : 2015/01/01           */
/*--------------------------------------*/
void overlaps( int n ) {
    int    i, j, k;
    int    ans;
    int    wk_ans;
    char   str[20];                 /* dataset(string) from stdio */
    double wk_x;
    double wk_y;
    double wk_dist;
    struct StdDataset dset[ n ];    /* dataset from stdio */
    
    /* read datasets from stdio */
    printf( MSG_SCD );
    for( i = 0; i < n; i++ ) {
        scanf( "%s", str );
        mystrtod( str, &wk_x, &wk_y );
        
        /* input value check */
        if ( NUM_MAX_XY < wk_x || NUM_MIN_XY > wk_x
         ||  NUM_MAX_XY < wk_y || NUM_MIN_XY > wk_y ) { /* error */
            printf( MSG_ERR_XY );
            i--;
        }
        else {                                          /* ok    */
            dset[ i ].x = wk_x;
            dset[ i ].y = wk_y;
        }
    }
    
    ans     = 1;
    wk_dist = 0;
    
    if( 1 >= n) {
        /* no process */
    }
    else {
        for (j = 0; j < n - 1; j++ ) {
            wk_ans = 1;
            for (k = j + 1; k < n; k++ ) {
                wk_dist = dist( dset[ j ].x, dset[ k ].x, dset[ j ].y, dset[ k ].y );
                if( NUM_DIST >= wk_dist ) {
                    wk_ans++;
                }
                else {
                    /* no process */
                }
            }
            if ( wk_ans > ans ) {
                ans = wk_ans;
            }
            else {
                /* no process */
            }
        }
    }
    printf("%d", ans);
    printf("\n");
}

/*--------------------------------------*/
/* Function Name : main                 */
/* Argument      : int argc             */
/*                 const char * argv[]  */
/* Output        : -                    */
/* Author        : slangsoft            */
/* Creation Date : 2015/01/01           */
/*--------------------------------------*/
int main( int argc, const char * argv[] ) {
    int    max_n;   /* number of datasets from stdio    */
    
    /* read number of datasets from stdio */
    printf( MSG_FST );
    scanf( "%d", &max_n );
    
    /* input value check */
    while( NUM_EXIT != max_n ) {
        if( NUM_MAX_N < max_n ) {   /* error */
            printf( MSG_ERR_N );
            return -1;
        }
        else {                      /* ok    */
            overlaps( max_n );
            printf( MSG_FST );
            scanf( "%d", &max_n );
        }
    }
    
    return 0;
}

File: Volume0_0090/main.php
<?php
//
// main.php
// Volume0_0090
//
// Created by slangsoft on 2015/01/23.
// Copyright (c) 2015 Slang Software. All rights reserved.
//

define( EOSTDIN , "0" );
define( NUM_DIST, 2   );

$strStdin   = "";
$maxN       = 0;        // number of datasets from stdin
$arrDataset = array();
$ans        = 1;
$wkAns      = 1;

// read number of datasets from stdin
while( $strStdin != EOSTDIN ) {
    $strStdin = trim( fgets( STDIN ) );
    list( $x, $y, $dummy ) = explode( ",", $strStdin, 3 );
    $arrDataset[] = array( "x" => floatval( $x ), "y" => floatval( $y ) );
}

// get the number of datasets
$maxN = $arrDataset[0]["x"];

// remove the first element from the array
array_splice( $arrDataset, 0, 1 );

// remove the last element from the array
array_pop( $arrDataset );
reset( $arrDataset );

if( 0 < $maxN ) {
    for($i = 0; $i <= $maxN - 2; $i++ ) {
        $wkAns = 1;
        for( $j = $i + 1; $j <= $maxN - 1; $j++ ) {
            $dist = new dist();
            $dist->setCenterPointA( $arrDataset[$i]["x"], $arrDataset[$i]["y"] );
            $dist->setCenterPointB( $arrDataset[$j]["x"], $arrDataset[$j]["y"] );
            if( NUM_DIST >= $dist->getDist() ) {
                $wkAns++;
            }
        }
        if( $wkAns > $ans ) {
            $ans = $wkAns;
        }
    }
} else {
    $ans = 0;
}

// output answer to stdout
fwrite( STDOUT, "Ans:" . $ans . PHP_EOL );

class dist {
    private $a, $b;
    private $x1, $x2, $y1, $y2;
    private $dist;

    public function getDist() {
        $this->x1 > $this->x2 ? $this->a = $this->x1 - $this->x2 : $this->a = $this->x2 - $this->x1;
        $this->y1 > $this->y2 ? $this->b = $this->y1 - $this->y2 : $this->b = $this->y2 - $this->y1;
        $this->dist = sqrt( pow( $this->a, 2 ) + pow( $this->b, 2 ) );
        return $this->dist;
        
    }
    public function setCenterPointA( $x, $y ) {
        $this->x1 = $x;
        $this->y1 = $y;
    }
    public function setCenterPointB( $x, $y ) {
        $this->x2 = $x;
        $this->y2 = $y;
    }
}
?>